onvif/scripts/zmonvif-probe.pl

```diff
@@ -95,7 +95,9 @@ sub deserialize_message
 
 sub interpret_messages
 {
-  my ($svc_discover, @responses, %services) = @_;
+  my ($svc_discover, $responses_ref, $services_ref) = @_;
+  my @responses = @$responses_ref;
+  my %services = %$services_ref;
```

Comment:
Are you sure that this doesn't create a copy of the hash, preventing it from being used to suppress duplicate responses between the 1.1 and 1.2 discovery phases?